nix/tools/generate_targets.py

```diff
@@ -0,0 +1,94 @@
+# generates a file to be used by the vegeta load testing tool
+import time
+import hmac
+import hashlib
+import base64
+import json
+import argparse
+import sys
+import random
+
+SECRET = b"reallyreallyreallyreallyverysafe"
+URL = "http://postgrest"
+JWT_DURATION = 120
+TOTAL_TARGETS = 50000  # tuned by hand to reduce result variance
+
+
+def base64url_encode(data: bytes) -> str:
+    """URL-safe Base64 encode without padding."""
+    return base64.urlsafe_b64encode(data).rstrip(b"=").decode("ascii")
+
+
+def generate_jwt(exp_inc: int) -> str:
+    """Generate an HS256 JWT"""
+    # Header & payload
+    header = {"alg": "HS256", "typ": "JWT"}
+    now = int(time.time())
+    payload = {
+        "sub": f"user_{random.getrandbits(32)}",
+        "iat": now,
+        "exp": now + exp_inc,
+        "role": "postgrest_test_author",
+    }
+
+    # Encode to JSON and then to Base64URL
+    header_b = json.dumps(header, separators=(",", ":")).encode()
+    payload_b = json.dumps(payload, separators=(",", ":")).encode()
+    header_b64 = base64url_encode(header_b)
+    payload_b64 = base64url_encode(payload_b)
+
+    # Sign (HMAC‑SHA256) the "<header>.<payload>" string
+    signing_input = f"{header_b64}.{payload_b64}".encode()
+    signature = hmac.new(SECRET, signing_input, hashlib.sha256).digest()
+    signature_b64 = base64url_encode(signature)
+
+    return f"{header_b64}.{payload_b64}.{signature_b64}"
+
+
+# We want to ensure 401 Unauthorized responses don't happen during
+# JWT validation, this can happen when the jwt `exp` is too short.
+# At the same time, we want to ensure the `exp` is not too big,
+# so expires will occur and postgREST will have to clean cached expired JWTs.
+def estimate_adequate_jwt_exp_increase(iteration: int) -> int:
+    # estimated time takes to build and run postgrest itself
+    build_run_postgrest_time = 2
+    # estimated time it takes to generate the targets file
+    file_generation_time = TOTAL_TARGETS//(10**-5)
+    # estimated exp time so some JWTs will expire
+    dynamic_exp_inc = iteration//1000
+
+    return build_run_postgrest_time + file_generation_time + dynamic_exp_inc
```

Comment:
This is the main addition in comparison to https://github.com/PostgREST/postgrest/pull/4023

Reply:
The clock skew is only honored at the time of insertion into cache. We don't take it into account when we lookup the cache.

Reply:
@taimoorzaeem That looks like the right behavior.

So the clock skew is only considered at the JWT validation phase but not during cache invalidation phase.

Reply:
>  I'm not sure whether that's the correct thing to do, though.

Yeah on second thought, doesn't make sense to slow down requests from "still valid" JWT caches.

Reply:
I wonder if we should make the skew configurable, so we can output it with `dump-config` then we could also use it as parameter of this loadtest.

That's something that can be done in another PR though.

Reply:
> I wonder if we should make the skew configurable, so we can output it with dump-config then we could also use it as parameter of this loadtest.

Opened https://github.com/PostgREST/postgrest/issues/4035